source/Ship.cpp

```diff
@@ -1733,15 +1757,15 @@ void Ship::DoGeneration()
 // Launch any ships that are ready to launch.
 void Ship::Launch(list<shared_ptr<Ship>> &ships, vector<Visual> &visuals)
 {
-	// Allow fighters to launch from a disabled ship, but not from a ship that
+	// Allow carried ships to launch from a disabled ship, but not from a ship that
 	// is landing, jumping, or cloaked. If already destroyed (e.g. self-destructing),
 	// eject any ships still docked, possibly destroying them in the process.
 	bool ejecting = IsDestroyed();
 	if(!ejecting && (!commands.Has(Command::DEPLOY) || zoom != 1.f || hyperspaceCount || cloak))
 		return;
 	
 	for(Bay &bay : bays)
-		if(bay.ship && ((bay.ship->Commands().Has(Command::DEPLOY) && !Random::Int(40 + 20 * bay.isFighter))
+		if(bay.ship && ((bay.ship->Commands().Has(Command::DEPLOY) && !Random::Int(40 + 20 * (crew > 0)))
```

Comment:
This makes carrier ships with a crew > 0 launch any of its carried ships more slowly than a carrier ship that is an automaton, even if the carried ships themselves are not automatons.

While that's a neat idea (i.e. an automaton carrier issues the launch order faster than a non-automaton carrier), the current behavior is that a drone "pilot" responds to the order faster than a live pilot.

I think you should be checking `bay.ship->attributes.Get("automaton")`

Reply:
My original idea was to use bay.ship->crew, so this was actually a programming error (taking the crew from the carrier instead of the carried ship). But `bay.ship->attributes.Get("automaton")` also works nicely, so using that one.